utils/common/src/main/java/brooklyn/util/time/Time.java

```diff
@@ -443,36 +500,466 @@ else if (s.equalsIgnoreCase("d") || s.equalsIgnoreCase("day") || s.equalsIgnoreC
+            if ("CDT".equals(code)) return getTimeZone("America/Chicago");
+            if ("MST".equals(code)) return getTimeZone("America/Denver");
+            if ("MDT".equals(code)) return getTimeZone("America/Denver");
+
+            if ("BST".equals(code)) return getTimeZone("Europe/London");  // otherwise BST is Bangladesh!
+            if ("CEST".equals(code)) return getTimeZone("Europe/Paris");
+            // IST falls through to below, where it is treated as India (not Irish); IDT not recognised
+        }
+        
+        TimeZone tz = TimeZone.getTimeZone(code);
+        if (tz!=null && !tz.equals(TimeZone.getTimeZone("GMT"))) {
+            // recognized
+            return tz;
+        }
+        // possibly unrecognized -- GMT returned if not known, bad TimeZone API!
+        String timeZones[] = TimeZone.getAvailableIDs();
+        for (String tzs: timeZones) {
+            if (tzs.equals(code)) return tz;
+        }
+        // definitely unrecognized
+        return null;
+    }
+    
+    /** convert a TimeZone e.g. Europe/London to an offset string as at the given day, e.g. +0100 or +0000 depending daylight savings,
+     * absent with nice error if zone unknown */
+    public static Maybe<String> getTimeZoneOffsetString(String tz, int year, int month, int day) {
+        TimeZone tzz = getTimeZone(tz);
+        if (tzz==null) return Maybe.absent("Unknown time zone code: "+tz);
+        return Maybe.of(getTimeZoneOffsetString(tzz, year, month, day));
+    }
+    
+    /** as {@link #getTimeZoneOffsetString(String, int, int, int)} where the {@link TimeZone} is already instantiated */
+    @SuppressWarnings("deprecation")
+    public static String getTimeZoneOffsetString(TimeZone tz, int year, int month, int day) {
+        int tzMins = tz.getOffset(new Date(year, month, day).getTime())/60/1000;
+        String tzStr = (tzMins<0 ? "-" : "+") + Strings.makePaddedString(""+(Math.abs(tzMins)/60), 2, "0", "")+Strings.makePaddedString(""+(Math.abs(tzMins)%60), 2, "0", "");
+        return tzStr;
+    }
+
+    private static String namedGroup(String name, String pattern) {
+        return "(?<"+name+">"+pattern+")";
+    }
+    private static String anyChar(String charSet) {
+        return "["+charSet+"]";
+    }
+    private static String optionally(String pattern) {
+        return "("+pattern+")?";
+    }
+    private static String options(String ...patterns) {
+        return "("+Strings.join(patterns,"|")+")";
+    }
+    private static String notMatching(String pattern) {
+        return "(?!"+pattern+")";
+    }
+    
+    private static Maybe<Matcher> match(String pattern, String input) {
+        Matcher m = Pattern.compile(pattern).matcher(input);
+        if (m.find() && m.start()==0 && m.end()==input.length())
+            return Maybe.of(m);
+        return Maybe.absent();
+    }
+
+//    // TODO https://github.com/joestelmach/natty is cool, but it drags in ANTLR,
```

Comment:
What is the TODO? To decide whether ANTLR is useful or not? Would prefer to either include Natty get rid of this commented block.